partiql-planner/src/main/kotlin/org/partiql/planner/typer/PlanTyper.kt

```diff
@@ -272,7 +272,8 @@ internal class PlanTyper(
          */
         override fun visitRexOpPath(node: Rex.Op.Path, ctx: StaticType?): Rex = rewrite {
             // 1. Resolve path prefix
-            val (root, steps) = when (val rootOp = node.root.op) {
+            val visitedRoot = visitRex(node.root, ctx)
```

Comment:
Nice catch, but I'm concerned this would cause problems because we might resolve the root accidentally when we should have considered the whole path like line 279.

By visiting the root first we might accidentally resolve to some global `x` rather than something like `x.y.z`. I don't have an example at the moment, but I think we need to consider the whole path for resolution in 280